goal_src/jak2/engine/gfx/texture/texture-anim.gc

```diff
@@ -6,37 +6,1325 @@
+                               (uv (new 'static 'gs-uv :v #x8 :u (+ (* t0-7 256) 8)))
+                               (xyz2 (new 'static 'gs-xyz :x (+ (* t0-7 256) 128)))
+                               (uv (new 'static 'gs-uv :u (+ (* t0-7 256) 136) :v (+ (* a3-4 16) 8)))
+                               (xyz2 (new 'static 'gs-xyz :y (* a3-4 16) :x (+ (* t0-7 256) 256)))
+                               )
+        )
+      )
+    (let ((t1-40 (/ width 64)))
+      (dma-buffer-add-gs-set dma-buf
+                             (frame-1 (new 'static 'gs-frame :psm (gs-psm ct16) :fbmsk fbmask :fbw tbw :fbp (+ fbp-to-draw t1-40)))
+                             (tex0-1 (new 'static 'gs-tex0 :psm #x2 :tcc #x1 :tfx #x1 :th th :tw tw :tbw tbw :tbp0 (+ tbp (* t1-40 32))))
+                             (prim (new 'static 'gs-prim :prim (gs-prim-type sprite) :tme #x1 :abe #x1 :fst #x1))
+                             )
+      )
+    )
+  (dotimes (v1-8 (/ width 16))
+    (dma-buffer-add-gs-set dma-buf
+                           (uv (new 'static 'gs-uv :v #x8 :u (+ (* v1-8 256) 8)))
+                           (xyz2 (new 'static 'gs-xyz :x (+ (* v1-8 256) 128)))
+                           (uv (new 'static 'gs-uv :u (+ (* v1-8 256) 136) :v (+ (* height 16) 8)))
+                           (xyz2 (new 'static 'gs-xyz :y (* height 16) :x (+ (* v1-8 256) 256)))
+                           )
+    )
+
+  ;; manually done from the stuff below.
+  (dma-buffer-add-gs-set-flusha dma-buf
+                                (xyoffset-1 (new 'static 'gs-xy-offset :ofx #x7000 :ofy #x7300))
+                                (frame-1 (new 'static 'gs-frame :fbw (/ (+ width 63) 64) :fbp fbp-to-draw))
+                                )
+  #|
+  ;; not sure why this doesn't become a macro
+  (let* ((v1-11 dma-buf)
+         (a0-15 (the-as dma-packet (-> v1-11 base)))
+         )
+    (set! (-> a0-15 dma) (new 'static 'dma-tag :qwc #x3 :id (dma-tag-id cnt)))
+    (set! (-> a0-15 vif0) (new 'static 'vif-tag :cmd (vif-cmd flusha) :msk #x1))
+    (set! (-> a0-15 vif1) (new 'static 'vif-tag :imm #x3 :cmd (vif-cmd direct) :msk #x1))
+    (set! (-> v1-11 base) (the-as pointer (&+ a0-15 16)))
+    )
+  (let* ((v1-12 dma-buf)
+         (a0-17 (the-as gs-gif-tag (-> v1-12 base)))
+         )
+    (set! (-> a0-17 tag) (new 'static 'gif-tag64 :nloop #x1 :eop #x1 :nreg #x2))
+    (set! (-> a0-17 regs) GIF_REGS_ALL_AD)
+    (set! (-> v1-12 base) (the-as pointer (&+ a0-17 16)))
+    )
+  (let ((v1-13 (-> dma-buf base)))
+    (set! (-> (the-as (pointer gs-xy-offset) v1-13) 0) (new 'static 'gs-xy-offset :ofx #x7000 :ofy #x7300))
+    (set! (-> (the-as (pointer gs-reg64) v1-13) 1) (gs-reg64 xyoffset-1))
+    (set! (-> (the-as (pointer gs-frame) v1-13) 2)
+          (new 'static 'gs-frame :fbw (/ (+ width 63) 64) :fbp fbp-to-draw)
+          )
+    (set! (-> (the-as (pointer gs-reg64) v1-13) 3) (gs-reg64 frame-1))
+    (set! (-> dma-buf base) (&+ v1-13 32))
+    )
+    |#
+  0
```

Comment:
make sure there's NO casts at all. The detection only works on the uncast (broken) output.

Reply:
I unfortunately couldn't get it to work. The variables are used a little differently, and it wasn't an easy fix to make this one work.